xpdan/tools.py

```diff
@@ -0,0 +1,169 @@
+import numpy as np
+from matplotlib.path import Path
+import scipy.stats as sts
+
+
+def margin(img_shape, edge_size):
+    """
+    Mask the edge of an image
+
+    Parameters
+    -----------
+    img_shape: tuple
+        The shape of the image
+    edge_size: int
+        Number of pixels to mask from the edge
+
+    Returns
+    --------
+    2darray:
+        The mask array, bad pixels are 0
+    """
+    mask = np.ones(img_shape, dtype=bool)
+    mask[edge_size:-edge_size, edge_size:-edge_size] = 0.
+    return ~mask
+
+
+def binned_outlier(img, r, alpha, bins, mask=None):
+    """
+    Generates a mask by identifying outlier pixels in bins and masks any
```

Comment:
I like to make a one-line summary, then an open line, then a longer description.  this works well with python auto-documentation.  The first line is a global statement of what the function does without details (and less than 80 chars long of course).  I think it has to be on teh same line as the opening """ though not sure about that.